### PR TITLE
Implement ldb list_live_files_metadata

### PR DESCRIPTION
**Pull request by bjlemaire**
Add an argument to ldb to dump live file names, column families, and levels, `list_live_files_metadata`. The output shows all active SST file names, sorted first by column family and then by level. For each level the SST files are sorted alphabetically.

Typically, the output looks like this:
```
./ldb --db=/tmp/test_db list_live_files_metadata
Live SST Files:
===== Column Family: default =====
---------- level 0 ----------
/tmp/test_db/000069.sst
---------- level 1 ----------
/tmp/test_db/000064.sst
/tmp/test_db/000065.sst
/tmp/test_db/000066.sst
/tmp/test_db/000071.sst
---------- level 2 ----------
/tmp/test_db/000038.sst
/tmp/test_db/000039.sst
/tmp/test_db/000052.sst
/tmp/test_db/000067.sst
/tmp/test_db/000070.sst
------------------------------
```

Second, a flag was added `--sort_by_filename`, to change the layout of the output. When this flag is added to the command, the output shows all active SST files sorted by name, in front of which the LSM level and the column family are mentioned. With the same example, the following command would return:
```
./ldb --db=/tmp/test_db list_live_files_metadata --sort_by_filename
Live SST Files:
/tmp/test_db/000038.sst : level 2, column family 'default'
/tmp/test_db/000039.sst : level 2, column family 'default'
/tmp/test_db/000052.sst : level 2, column family 'default'
/tmp/test_db/000064.sst : level 1, column family 'default'
/tmp/test_db/000065.sst : level 1, column family 'default'
/tmp/test_db/000066.sst : level 1, column family 'default'
/tmp/test_db/000067.sst : level 2, column family 'default'
/tmp/test_db/000069.sst : level 0, column family 'default'
/tmp/test_db/000070.sst : level 2, column family 'default'
/tmp/test_db/000071.sst : level 1, column family 'default'
------------------------------
```

 Thus, the user can either request to show the files by levels, or sorted by filenames.
This PR includes a simple Python unit test that makes sure the file name and level printed out by this new feature matches the one found with an existing feature, `dump_live_file`.

### CODE REVIEW

**Comment by akankshamahajan15** [comment]
Can you share how the output will be printed on the screen in the description. May be under the test plan.

File: tools/ldb_cmd.cc [hunk 1]
```diff
@@ -3387,6 +3391,99 @@ void DBFileDumperCommand::DoCommand() {
...
+  db_->GetLiveFilesMetaData(&metadata);
+  if (sort_by_filename_) {
+    std::sort(metadata.begin(), metadata.end(),
+              [](const LiveFileMetaData& a, const LiveFileMetaData& b) -> bool {
+                std::string aName = a.db_path + a.name;
+                std::string bName = b.db_path + b.name;
+                return (aName.compare(bName) < 0);
+              });
+    for (auto& fileMetadata : metadata) {
+      std::string filename = fileMetadata.db_path + fileMetadata.name;
```
Comment (anand1976) [inline tools/ldb_cmd.cc:3431]: Do we need to check for and strip out a trailing '/' here from ```fileMetadata.db_path```?

File: tools/ldb_test.py [hunk 2]
```diff
@@ -425,6 +425,40 @@ def testDumpLiveFiles(self):
...
 
+    def listLiveFilesMetadata(self, params, dumpFile):
+        return 0 == run_err_null("./ldb list_live_files_metadata %s > %s" % (
+            params, dumpFile))
+
+    def testListLiveFilesMetadata(self):
+        print("Running testListLiveFilesMetadata...")
+
+        dbPath = os.path.join(self.TMP_DIR, self.DB_NAME)
+        self.assertRunOK("put x1 y1 --create_if_missing", "OK")
```
Comment (anand1976) [inline tools/ldb_test.py:436]: Can we add a couple of column families to exercise that path?

**Review by anand1976 — COMMENTED** [review]
Looks great! I've left a few comments - nothing major.

**Review by akankshamahajan15 — APPROVED** [review]
LGTM! History.md can be updated with newly added option.

**Review by anand1976 — APPROVED** [review]

File: HISTORY.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 * Subcompactions are now disabled when user-defined timestamps are used, since the subcompaction boundary picking logic is currently not timestamp-aware, which could lead to incorrect results when different subcompactions process keys that only differ by timestamp.
 
 ### New Features
+* ldb now has a new feature, `list_live_files_metadata`, which lists all the live SST files, their column families and the LSM level they are located at.
```
Comment (akankshamahajan15) [inline HISTORY.md:11]: Branch is outdated because 6.22 branch has been cut. You can add unreleased and New feature  section and update your change there.

Comment (bjlemaire) [reply HISTORY.md:11]: Oh I see this is subtle, thanks for the catch. There was already an "Unreleased" section I didn't realize that there was no 6.22 yet. Adding a new unreleased in the next push.

**Pull request closed without merging**